src/scippneutron/absorption/cylinder.py

```diff
@@ -0,0 +1,240 @@
+from dataclasses import dataclass
+from typing import Literal
+
+import numpy as np
+import scipp as sc
+from numpy.polynomial.chebyshev import chebgauss
+from numpy.polynomial.legendre import leggauss
+
+from . import quadratures
+from .types import SampleShape
+
+
+@dataclass
+class Cylinder(SampleShape):
+    symmetry_line: sc.Variable
+    center_of_base: sc.Variable
+    radius: sc.Variable
+    height: sc.Variable
+
+    def beam_intersection(
+        self, start_point: sc.Variable, direction: sc.Variable
+    ) -> sc.Variable:
+        'Length of intersection between beam and cylinder'
```

Comment:
```suggestion
"""Length of intersection between beam and cylinder"""
```
I'm surprised this isn't caught by ruff.